Add documentation for Iceberg incremental MV refresh (#22549)
trinodb/trino: docs/src/main/sphinx/connector/iceberg.md

    `query_partition_filter_required` catalog session property for temporary,
    catalog specific use.
  - `false`
* - `iceberg.incremental-refresh-enabled`
  - Set to `false` to force the materialized view refresh operation to always perform a full refresh. You can use the
    `incremental_refresh_enabled` catalog session property for temporary, catalog specific use. 
    In the majority of cases, using incremental refresh (vs full refresh) will be more beneficial since a much smaller 

>     subset of the source table(s) needs to be scanned. One caveat is that while incremental refresh may scan less data, 

Reviewer: ```suggestion
    subset of the source tables needs to be scanned. While incremental refresh may scan less data, 
```

>     it may result in more data files created since it uses the append operation to insert the 'new' records.

Reviewer: ```suggestion
    it may result in the creation of more data files, since it uses the append operation to insert the new records.
```

must run {doc}`/sql/refresh-materialized-view` to populate data in the
materialized view.

Updating the data in the materialized view can be achieved using the `REFRESH MATERIALIZED VIEW` command.
This operation may perform either an incremental or a full refresh, depending on the complexity of the materialized view 

> definition and the snapshot history of the source table(s). For full refresh, the operation deletes the data 

Reviewer: ```suggestion
definition and the snapshot history of the source tables. For a full refresh, the operation deletes the data 
```

from the storage table, and inserts the data that is the result of executing the materialized view query into the 

> existing table. For incremental refresh, the existing data is not deleted from the storage table and only the 'delta' 

Reviewer: ```suggestion
existing table. For incremental refresh, the existing data is not deleted from the storage table and only the delta
```

Reviewer: Or maybe "only the changed records" .. instead of the technical "delta" term

> records are processed from the source table(s) and appended into the storage table (if needed).

Reviewer: ```suggestion
records are processed from the source tables and appended into the storage table as needed.
```

> In both cases, data is replaced/appended atomically, so users can continue to query the materialized view while

Reviewer: ```suggestion
In both cases, data is replaced or appended atomically, so users can continue to query the materialized view while
```
